styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -476,18 +478,19 @@ private boolean isNonDeletePeriodExpired(ContainerStatus cs) {
       LOG.warn("Failed to parse container state terminated finishedAt: '{}'", t.getFinishedAt(), e);
       return true;
     }
-    final Instant deadline = time.get().minus(Duration.ofSeconds(podDeletionDelaySeconds));
+    final Instant deadline = time.get().minus(podDeletionDelay);
     return finishedAt.isBefore(deadline);
   }
 
   @VisibleForTesting
-  void deletePod(WorkflowInstance workflowInstance, Pod pod, String reason) {
+  boolean deletePod(WorkflowInstance workflowInstance, Pod pod, String reason) {
```

Comment:
This method name suggest an action: delete the pod, but now the implementation just return a boolean to signal if we should delete the pod or not. A better name for this method would be then `shouldDeletePod`.

Probably the same could be applied to the chain of methods that call `deletePod'

Reply:
👍